rasa/core/utils.py

```diff
@@ -460,15 +460,14 @@ def _lock_store_is_redis_lock_store(
         return False
 
     # `lock_store` is `None` or `EndpointConfig`
-    return lock_store is not None and lock_store.type == "redis"
+    return lock_store is not None and lock_store.type != "in_memory"
```

Comment:
It would be better to create a separate function called `_lock_store_is_custom_lock_store` to handle this case. You'll need to update `number_of_sanic_workers` as well to call the function.